crossover4DBA.py

```diff
@@ -0,0 +1,269 @@
+      for fb in fragments_B:
+        new_mol_trial.append(rxn.RunReactants((fa,fb))[0]) 
+                                 
+    new_mols = []
+    for mol in new_mol_trial:
+      mol = mol[0]
+      if mol_OK(mol):
+        new_mols.append(mol)
+    
+    if len(new_mols) > 0:
+      return random.choice(new_mols)
+    
+  return None
+
+def linker_OK(mol,pos):
+    j = 0
+    if pos == 'middle':
+        for atom in mol.GetAtoms():
+            if atom.GetAtomicNum() == 0 and atom.GetIsotope() == 96:
+                j += 1
+            if atom.GetAtomicNum() == 0 and atom.GetIsotope() == 97:
+                j += 1
+        if j == 2:
+            return True
+        elif j < 2:
+            return False
+
+    elif pos == 'glcyl':
+        for atom in mol.GetAtoms():
+            if atom.GetAtomicNum() == 0 and atom.GetIsotope() == 98:
+                for neigh in atom.GetNeighbors():
+                    if neigh.GetAtomicNum() == 6:
+                    #print(neigh.GetIsAromatic(),neigh.IsInRing())
+                        if neigh.GetIsAromatic() == True and neigh.IsInRing() == True:
+                            j += 1
+            if atom.GetAtomicNum() == 0 and atom.GetIsotope() == 97:
+                j += 1
+        if j == 2:
+            return True
+        elif j < 2:
+            return False
+
+    elif pos == 'frcyl':
+        for atom in mol.GetAtoms():
+            if atom.GetAtomicNum() == 0 and atom.GetIsotope() == 99:
+                for neigh in atom.GetNeighbors():
+                    if neigh.GetAtomicNum() == 6:
+                    #print(neigh.GetIsAromatic(),neigh.IsInRing())
+                        if neigh.GetIsAromatic() == True and neigh.IsInRing() == True:
+                            j += 1
+            if atom.GetAtomicNum() == 0 and atom.GetIsotope() == 96:
+                j += 1
+        if j == 2:
+            return True
+        elif j < 2:
+            return False
+
+def DBA_crossover_OK(mol):
+    i = 0 
+    j = 0
+    dummies = []
+    for atom in mol.GetAtoms():
+        if atom.GetAtomicNum() == 0 and atom.GetIsotope() == 98:
+            dummies.append(atom.GetIdx())
+            #print('Atom idx and N:',atom.GetAtomicNum(),atom.GetIdx())
+            for neigh in atom.GetNeighbors():
+                if neigh.GetAtomicNum() == 6:
+                    if neigh.GetIsAromatic() == True and neigh.IsInRing() == True:
+                        i += 1
+                        #print('Value of i =',i)
+        if atom.GetAtomicNum() == 0 and atom.GetIsotope() == 99:
+            dummies.append(atom.GetIdx())
+            #print('Atom idx and N:',atom.GetAtomicNum(),atom.GetIdx())
+            for neigh in atom.GetNeighbors():
+                if neigh.GetAtomicNum() == 6:
+                    if neigh.GetIsAromatic() == True and neigh.IsInRing() == True:
+                        j += 1
+                        #print('Value of j =',j)
+
+    #print('Dummies',len(dummies))
```

Comment:
Here add a try except to check if the tweezer can be classified by tweezer_clasification. Continue if try works, return None if except raise.

Reply:
better  include the classification inside the i and j counters. If the tweezer can be classified as 0, 1 or 2, I can verfify if the dummy is bonded to a bencene ring.